lib/ObjWriter/objwriter.cpp

```diff
@@ -249,6 +252,56 @@ extern "C" void FinishObjWriter(ObjectWriter *OW) {
   delete OW;
 }
 
+extern "C" void CreateDataSection(ObjectWriter *OW, const char *SectionName, bool ReadOnly) {
+  assert(OW && "ObjWriter is null");
+  Triple TheTriple(TripleName);
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = *AsmPrinter->OutStreamer;
+  MCContext &OutContext = OST.getContext();
+
+  std::string SectionNameStr(SectionName);
+  assert(OW->CustomSections.find(SectionNameStr) != OW->CustomSections.end() && 
+    "Section with duplicate name already exists");
+
+  MCSection *Section = nullptr;
+  SectionKind Kind = ReadOnly ? 
+      SectionKind::getReadOnly() : SectionKind::getData();
+
+  switch (TheTriple.getObjectFormat()) {
+    case Triple::MachO:
+      Section = OutContext.getMachOSection("__DATA", SectionName, 0, Kind);
+      break;
+    case Triple::COFF:
+      {
```

Comment:
Braces are off here and in the if/else statements throughout. As per the LLVM coding standards, opening braces should be on the same line as the construct (e.g. `if (...) {`, `case Foo: {`, etc.).